src/main/java/frc/robot/Splash.java

```diff
@@ -58,25 +58,32 @@ private static void printStatusFile(
       System.out.print((filename + ": ").replace(".txt", ""));
 
       try {
-        for (int length = 0; (length = statusfile.read(buffer)) != -1; ) {
-          String buf =
-              new String(buffer, StandardCharsets.UTF_8).replaceAll("\\s", " ").replace("\0", "");
-          String tfn = filename.replace(".txt", "");
-          String fn = tfn.substring(0, 1).toUpperCase(Locale.ENGLISH) + tfn.substring(1);
-          System.out.write(buffer, 0, length);
-          SmartDashboard.putString(fn, buf);
-          tab = Shuffleboard.getTab("Status");
-          field =
-              tab.add(fn, buf).withPosition(colIndex, rowIndex).withSize(widthIndex, 1).getEntry();
-          field.setString(buf);
+        if (statusfile == null) {
+          System.out.print("File not found.");
```

Comment:
This is just a temporary bandage, and it is not fixing the root cause of the problem. Have we tested this if this change won't affect on the running robot.

Reply:
It isn't a bandage. The cause of the problem is in the VSCode build system, and not in the code. The null will never be returned on a running robot because the robot code is always run from a jar file, while in VSCode it isn't. This isn't documented anywhere but the WPI developers have run into similar issues before and confirmed it. Since we can't guarantee that statusfile will not be null (except on the robot) the correct way to handle it is with a null check.

The addition of the placeholder file is only marginally related, since there was also an issue with gradle not being aware of the resources until after the first build. 

This code has obviously not been tested on the robot, but it has been tested in the simulator as is, also with errors injected. Since the complaint was that this issue was slowing development, I would suggest integrating it right away. But I will leave that decision to you.

Reply:
I would like to ask to defer this. We will keep an eye out on the simulation environment.

Reply:
I am confused. You said this problem was a major issue slowing development.

Reply:
It was an issue for us early on when we didn't have the test board and we tried to test everything in the simulator. We have tried debugging the issue and we ended up moving the project folder around. It appeared to solved the issue, but sometime the issue came back. I think that I would like to under better on the root cause first. We are in competition seasons that I'd like focus on getting the working code on the robot.

Reply:
That's fine. We can hold off. But I understand the issue now. The getResourceAsStream method is supposed to return the file designated from the java resources, which is usually in the jar file. However, the VSCode debug environment doesn't use the Jar file, it uses the class files in place under the build directory. So there is a designated directory where the resources in the source tree get copied to, but the copy is only done if necessary. The logic behind the term "if necessary" is faulty. The first time the build is run in the simulator in a fresh workspace, the logic tests for the existence of resources before the part of the build that creates them has run, so it thinks there are no resources to copy. On subsequent builds, the resources folder's timestamp hasn't changed, so the logic thinks there are no new resources to copy. No resources and the method returns null. However, there are certain actions, such as changing the Enable Desktop Support status, that cause the build to be done completely from scratch, which triggers the resources to be copied. Once they are copied the first time, the logic starts checking for updates.

That all being said, handling the null in the code like that is the correct solution regardless. Failure returns should always be checked and dealt with.